packages/viz/src/lib/Layer.ts

```diff
@@ -115,24 +108,35 @@ export class Layer {
     return this._deckLayer;
   }
 
+  private async _getLayerProperties() {
+    const metadata = this._source.getMetadata();
+
+    const styleProps = this._style
+      ? this._style.getProperties(this._source)
+      : undefined;
+
+    const props = this._source.getProps();
+
+    return {
```

Comment:
Order matters in this object, so I think we should change it a little bit.

My proposal is to put static properties first (`type` property) so that they can be overriden by dynamic ones below. 
After that I would put `props` from the source, and below that property I would put `this._options` just in case we need to override any properties from the outside.

Styles definition will be the ones below all the previous ones. The ordering in `defaultStyles` and `styleProps` is good.